src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -184,7 +184,7 @@ void IDeserializationCallback.OnDeserialization(object sender)
 		internal SessionImpl(SessionFactoryImpl factory, ISessionCreationOptions options)
 			: base(factory, options)
 		{
-			using (new SessionIdLoggingContext(SessionId))
+			using (BeginContext())
```

Comment:
This constructor does check status, but as its latest step. So just setting the session id.